lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,322 @@
+        try:
+            result = self.client.post(subcollection_url, resources=alerts, action=action)
+            if len(result['results']) != len(alerts):
+                msg = "Failed to {action} alerts to profile '{description}'," +\
+                      "expected {expected} alerts to be {action}ed," +\
+                      "but only {changed} were {action}ed"
+                msg = msg.format(action=action,
+                                 description=profile['description'],
+                                 expected=len(alerts),
+                                 changed=result['results'])
+                self.module.fail_json(msg=msg)
+        except Exception as e:
+            msg = "Failed to {action} alerts to profile '{description}': {error}"
+            msg = msg.format(action=action, description=profile['description'], error=e)
+            self.module.fail_json(msg=msg)
+
+        return result['results']
+
+    def update_profile(self, old_profile, desired_profile):
+        """ Update alert profile in ManageIQ
+        """
+        changed = False
+        # we need to use client.get to query the alert definitions
+        old_profile = self.client.get(old_profile['href'] + '?expand=alert_definitions')
+
+        # figure out which alerts we need to assign / unassign
+        # alerts listed by the user:
+        desired_alerts = set(self.get_alerts(desired_profile['alerts']))
+
+        # alert which currently exist in the profile
+        if 'alert_definitions' in old_profile:
+            # we use get_alert_href to have a direct href to the alert
+            existing_alerts = set([self.get_alert_href(alert) for alert in old_profile['alert_definitions']])
+        else:
+            # no alerts in this profile
+            existing_alerts = set()
+
+        to_add = list(desired_alerts - existing_alerts)
+        to_remove = list(existing_alerts - desired_alerts)
+
+        # assign / unassign the alerts, if needed
+
+        if to_remove:
+            self.assign_or_unassign(old_profile, to_remove, "unassign")
+            changed = True
+        if to_add:
+            self.assign_or_unassign(old_profile, to_add, "assign")
+            changed = True
+
+        # update other properties
+        profile_dict = dict()
+        if old_profile['description'] != desired_profile['description']:
+            # description needs to be updated
+            profile_dict['description'] = desired_profile['description']
+
+        if old_profile['mode'] != desired_profile['resource_type']:
+            # mode needs to be updated
+            profile_dict['mode'] = desired_profile['resource_type']
+
+        # check if notes need to be updated
+        old_notes = None
+        if 'set_data' in old_profile and 'notes' in old_profile['set_data']:
+            old_notes = old_profile['set_data']['notes']
+
+        if desired_profile['notes'] != old_notes:
+            # existing profile has no notes, but we have notes
```

Comment:
I think this comment is not precise now, this also happens when existing has some old_notes, new has empty string, or both old/new non-empty but different...  Can just drop it, or "# notes needs to be updated".